src/ops/validate.rs

```diff
@@ -513,9 +514,11 @@ mod test {
 
         let input_node = OpType::Dataflow(DataflowOp::Input {
             types: in_types.clone(),
+            resources: ResourceSet::new(),
```

Comment:
on main `Input` and `Output` are now standalone structs, so you can define constructors for them to make this easier.
I suggest 
```rust 
fn new(types: TypeRow) -> Self;
fn with_resources(self, resources: ResourceSet) -> Self;
```
 So you can do `ops::Input::new(type_row![QB]).with_resources(...)`

Reply:
Done!